core/includes/module.inc

```diff
@@ -576,6 +578,9 @@ function module_disable($module_list, $disable_dependents = TRUE) {
     $module_list = array_keys($module_list);
   }
 
+  // Refresh Entity info static cache.
+  entity_info_cache_clear();
```

Comment:
Is there a reason why this is here instead of 20 lines down, where all the other caches are cleared?

Reply:
The cache has to be cleared before `hook_disable()` is invoked in `module_invoke($module, 'disable')`.